### ISSUE TITLE
TouchArea has-hover property is not changing when being overlayed by other component.

### ISSUE THREAD

**Issue by szymoneiro** [slint-ui/slint#2716]
Here's an [example](https://slint-ui.com/releases/1.0.2/editor/?snippet=export+component+BugWindow+inherits+Window+%7B%0A++++width%3A+600px%3B%0A++++height%3A+600px%3B%0A%0A++++Rectangle+%7B%0A++++++++width%3A+100px%3B%0A++++++++height%3A+100px%3B%0A++++++++background%3A+area.has-hover+%3F+red+%3A+yellow%3B%0A++++++++x%3A+220px%3B%0A++++++++opacity%3A+0.5%3B%0A++++++++area+%3A%3D+TouchArea+%7B%7D%0A++++%7D%0A%0A++++Rectangle+%7B%0A++++++++width%3A+100px%3B%0A++++++++height%3A+100px%3B%0A++++++++background%3A+green%3B%0A++++++++opacity%3A+0.5%3B%0A++++++++x%3A+280px%3B%0A++++++++secondArea+%3A%3D+TouchArea+%7B%7D%0A++++%7D%0A%7D)
or code below:
```
export component BugWindow inherits Window {
    width: 600px;
    height: 600px;

    Rectangle {
        width: 100px;
        height: 100px;
        background: area.has-hover ? red : yellow;
        x: 220px;
        opacity: 0.5;
        area := TouchArea {}
    }

    Rectangle {
        width: 100px;
        height: 100px;
        background: green;
        opacity: 0.5;
        x: 280px;
        secondArea := TouchArea {}
    }
}
```
Basically when TouchArea is being overlayed with other component, then its has-hover property is not reacting properly when mouse is no longer over it.

**Comment by tronical** [comment]
[SlintPad link](https://slint-ui.com/releases/1.0.2/editor/?snippet=export+component+BugWindow+inherits+Window+%7B%0A++++width%3A+600px%3B%0A++++height%3A+600px%3B%0A%0A++++Rectangle+%7B%0A++++++++width%3A+100px%3B%0A++++++++height%3A+100px%3B%0A++++++++background%3A+area.has-hover+%3F+red+%3A+yellow%3B%0A++++++++x%3A+220px%3B%0A++++++++opacity%3A+0.5%3B%0A++++++++area+%3A%3D+TouchArea+%7B%7D%0A++++%7D%0A%0A++++Rectangle+%7B%0A++++++++width%3A+100px%3B%0A++++++++height%3A+100px%3B%0A++++++++background%3A+green%3B%0A++++++++opacity%3A+0.5%3B%0A++++++++x%3A+280px%3B%0A++++++++secondArea+%3A%3D+TouchArea+%7B%7D%0A++++%7D%0A%7D) 

@ogoffart when you have time: Do you agree that this looks like a bug?

**Comment by ogoffart** [comment]
Yes, this is clearly a bug

**Comment by Adancurusul** [comment]
I use this feature to achieve a similar pop-up overlay display, so that clicking on the upper layer will not accidentally touch the TouchArea below

![E_U27JHGZD}S@ _G4 7272H](https://github.com/slint-ui/slint/assets/54146249/027cd9af-a355-4de9-b495-1db69f5c3b4c)

![TWOU~9$XFX{${BG1Y@~OMNA](https://github.com/slint-ui/slint/assets/54146249/acb8d811-262d-4923-bd11-9a21c96c5ad8)
